client/state/sites/hooks/use-premium-global-styles.ts

```diff
@@ -41,34 +38,18 @@ const getGlobalStylesInfoForSite = ( siteId: number | null ): GlobalStylesStatus
 		.catch( () => DEFAULT_GLOBAL_STYLES_INFO );
 };
 
-export function usePremiumGlobalStyles(): GlobalStylesStatus {
-	const params = new URLSearchParams( window.location.search );
-	const siteSlugParam = params.get( 'siteSlug' );
-	const siteIdParam = params.get( 'siteId' );
-	const selectedSiteId = useSelector( getSelectedSiteId );
-	const onboard = select( ONBOARD_STORE ).getState();
-
-	// When site id is null it means that the site hasn't been created yet.
-	const siteId = useSelector( ( state ) => {
-		const siteIdOrSlug =
-			onboard?.stepProgress !== undefined ? siteIdParam ?? siteSlugParam : selectedSiteId;
-
-		if ( ! siteIdOrSlug ) {
-			return null;
-		}
-
-		const site = getSite( state, siteIdOrSlug );
-		return site?.ID ?? null;
-	} );
-
-	const { data } = useQuery(
+export function usePremiumGlobalStyles( siteId: number | null = null ): GlobalStylesStatus {
+	const { isLoading, data } = useQuery(
 		[ 'globalStylesInfo', siteId ],
 		() => getGlobalStylesInfoForSite( siteId ),
 		{
-			placeholderData: DEFAULT_GLOBAL_STYLES_INFO,
 			refetchOnWindowFocus: false,
 		}
 	);
 
-	return data ?? DEFAULT_GLOBAL_STYLES_INFO;
+	return {
+		...DEFAULT_GLOBAL_STYLES_INFO,
+		...data,
+		isFetchingGlobalStylesStatus: isLoading,
+	};
```

Comment:
With this change, we can remove lines: 34 through 37 to avoid the same operation twice once we have a response.